### PR TITLE
Enable New Search tests in CI

### PR DESCRIPTION
**Pull request by crisptrutski**
About time 😅

### CODE REVIEW

File: test/metabase/db/custom_migrations_test.clj [hunk 1]
```diff
@@ -78,34 +78,42 @@
...
-                 ;; stop the scheduler because the scheduler won't be started when migrations start
-                 (task/stop-scheduler!)
-                 (migrate!)
-                 ;; check the job and trigger are deleted
-                 (task/start-scheduler!)
-                 (testing "after the migration, the job and trigger are deleted"
-                   (is (nil? (qs/get-job (@#'task/scheduler) (jobs/key abandonment-emails-job-key))))
-                   (is (nil? (qs/get-trigger (@#'task/scheduler) (triggers/key abandonment-emails-trigger-key)))))))
-           (finally (task/stop-scheduler!))))))
+    (with-search-indexing-disabled!
```
Comment (qnkhuat) [inline test/metabase/db/custom_migrations_test.clj:90]: why it's hitting this test tho?

File: test/metabase/db/custom_migrations_test.clj
Comment (qnkhuat) [inline test/metabase/db/custom_migrations_test.clj:1]: how about we the fixtures and turn this off for all tests under this namespace?

```clojure
(use-fixtures :each (fn [thunk]
                      (mt/with-temporary-setting-values [experimental-fulltext-search-enabled false]
                        (thunk))))
```

Comment (crisptrutski) [reply test/metabase/db/custom_migrations_test.clj:1]: I think this comment is for a stale commit? I can't see what code it's referring to.

Comment (crisptrutski) [reply test/metabase/db/custom_migrations_test.clj:1]: How right, yeah let's do it

Comment (crisptrutski) [reply test/metabase/db/custom_migrations_test.clj:1]: This ended up infeasible as the linter forbids impure code in fixtures.

File: src/metabase/cmd/load_from_h2.clj [hunk 2]
```diff
@@ -32,4 +35,5 @@
   ([h2-filename]
    (let [h2-filename    (str h2-filename ";IFEXISTS=TRUE")
          h2-data-source (copy.h2/h2-data-source h2-filename)]
-     (copy/copy! :h2 h2-data-source (mdb/db-type) (mdb/data-source)))))
+     (copy/copy! :h2 h2-data-source (mdb/db-type) (mdb/data-source))
+     (search.index/reset-tracking!))))
```
Comment (piranha) [inline src/metabase/cmd/load_from_h2.clj:39]: this is a bit weird as it's "postgres" and not just generic index

Comment (crisptrutski) [reply src/metabase/cmd/load_from_h2.clj:39]: Fixed in the refactor

**Review by sansolovyov — APPROVED** [review]

**Pull request merged**